### PR TITLE
cpu: tms34020: Implement cvdxyl_b opcode

### PR DESCRIPTION
**Pull request by plaes**
Each `cvdxyl_b` call in Atronic software seems to be clustered together with `vcol` and `vfill_l` opcodes which are not yet implemented.

### CODE REVIEW

File: src/devices/cpu/tms34010/34010ops.hxx [hunk 1]
```diff
@@ -2245,7 +2246,34 @@ void tms34020_device::cvdxyl_a(uint16_t op)
...
-	logerror("020:cvdxyl_b\n");
+	// Convert destination XY address to linear
+	// (Y half of Rd x DPTCH) + (X half of Rd x PSIZE) + (A4 or B4) ->  Rd
+	XY *rd = &BREG_XY(DSTREG(op));
+	off_t dptch = DPTCH();
+	off_t offset = BREG(4);
+
+	off_t result = (rd->y * dptch) + (rd->x * (1 << m_pixelshift)) + offset;
+	// logerror("020:cvdxyl_b: PC=0x%08x: DADDR=%x, x=%x, y=%x, DPTCH=%d, PIXELSIZE=%d, OFFSET=%d, result: %x\n",
+	// m_pc, rd, rd->x, rd->y, dptch, (1 << m_pixelshift), offset, result);
```
Comment (cuavas) [inline src/devices/cpu/tms34010/34010ops.hxx:2257]: Why are you using `(rd->x * (1 << m_pixelshift))`?  Shouldn’t `(rd->x << m_pixelshift)` without the extra multiplication step work?

File: src/devices/cpu/tms34010/34010ops.hxx [hunk 2]
```diff
@@ -2245,7 +2246,34 @@ void tms34020_device::cvdxyl_a(uint16_t op)
...
+
+	DADDR() = result;
+
+	// Handle cycles related to pitch:
+	switch(population_count_32(dptch))
+	{
+		// power of 2: 2 clocks
```
Comment (cuavas) [inline src/devices/cpu/tms34010/34010ops.hxx:2264]: Please format new code like the existing code in the file, like `switch (...)` (space between flow control keyword and opening parenthesis of the controlling expression).

File: src/devices/cpu/tms34010/34010ops.hxx [hunk 3]
```diff
@@ -2245,7 +2246,34 @@ void tms34020_device::cvdxyl_a(uint16_t op)
...
 void tms34020_device::cvdxyl_b(uint16_t op)
 {
-	logerror("020:cvdxyl_b\n");
+	// Convert destination XY address to linear
+	// (Y half of Rd x DPTCH) + (X half of Rd x PSIZE) + (A4 or B4) ->  Rd
+	XY *rd = &BREG_XY(DSTREG(op));
+	off_t dptch = DPTCH();
+	off_t offset = BREG(4);
+
+	off_t result = (rd->y * dptch) + (rd->x * (1 << m_pixelshift)) + offset;
```
Comment (cuavas) [inline src/devices/cpu/tms34010/34010ops.hxx:2255]: Can these things be `const`?  I don’t think any of them should be modified after assignment.

File: src/devices/cpu/tms34010/34010ops.hxx [hunk 4]
```diff
@@ -185,6 +185,7 @@ void tms340x0_device::cpw_b(uint16_t op) { CPW(B); }
 	R##REG(DSTREG(op)) = DXYTOL(R##REG_XY(SRCREG(op)));     \
 	COUNT_CYCLES(3);                                \
 }
+// TODO: Handle cycles properly, currently we assume power of two
 void tms340x0_device::cvxyl_a(uint16_t op) { CVXYL(A); }
 void tms340x0_device::cvxyl_b(uint16_t op) { CVXYL(B); }
```
[the remark this answers is not in the document]
Comment (plaes) [reply src/devices/cpu/tms34010/34010ops.hxx:190]: I moved the comment on top of the `CVXYL(...)` macro.

**Pull request merged**